### PR TITLE
Forward a configurable targeted k-CFA depth to the analysis engine

### PR DESCRIPTION
**Pull request by khatchad**
Threads a configurable targeted k-CFA depth from the refactoring through to Ariadne's `PythonTensorAnalysisEngine` (#600), defaulting to `MODEL_FORWARD_CFA_DEPTH` (4).

## What It Does
- `EclipsePythonProjectTensorAnalysisEngine` gains a depth-forwarding constructor over the 0.50.0 3-arg `PythonTensorAnalysisEngine(List<File>, String, int)`.
- `HybridizeFunctionRefactoringProcessor` carries a `targetedCfaDepth` (getter/setter), defaulting to a public `DEFAULT_TARGETED_CFA_DEPTH` constant, and forwards it when building the engine.
- The evaluator reads the depth from the `edu.cuny.hunter.hybridize.eval.targetedCfaDepth` system property and emits it as a `targeted CFA depth` column in `results.csv`.
- The test harness can forward a chosen depth, mirroring how it forwards `inferInputSignatures`.

## Why Default 4
Confirmed via wala/ML#587: for a parameter bound to a model-forward output across multiple call sites, the shallow `DEFAULT_TARGETED_CFA_DEPTH` (2) collapses the contexts and leaks one call site's shape into another, an unsound single-shape result; depth 4 separates the contexts and the set gains the safe over-approximation. `testTargetedCfaDepthSoundness` pins this on `accuracy`'s `y_pred` in `neural_network.py`: `{(256, 10)}` at depth 2 (unsound) versus `{(256, 10), ?}` at depth 4 (sound). Parameters bound to direct call-site arguments (e.g. `NeuralNet.call`'s `x`) are depth-invariant.

## Deferred Follow-ups
- A UI wizard field for the depth (the shared `InputPage` base exposes no extension hook, and the field is not CI-testable).
- The broader model-forward corpus (gcn_proj/nlpgnn) as additional coverage.

Closes #600.

### CODE REVIEW

File: edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java [hunk 1]
```diff
@@ -539,6 +540,13 @@ public static void tearDown() {
 	 */
 	protected boolean inferInputSignatures;
 
+	/**
+	 * The targeted k-CFA depth the harness forwards to the analysis engine, defaulting to
+	 * {@link HybridizeFunctionRefactoringProcessor#DEFAULT_TARGETED_CFA_DEPTH} so the suite's behavior is unchanged. A test sets it via
+	 * {@link #setTargetedCfaDepth(int)} to analyze a fixture at a chosen depth (#600).
+	 */
```
[the remark this answers is not in the document]
Comment (khatchad) [reply edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java:547]: Fixed in e6c50cf1305809521d70cf0a15b727431f6b86a8: reworded to "defaulting to the refactoring's own `DEFAULT_TARGETED_CFA_DEPTH` so the harness does not override it," dropping the misleading "unchanged" phrasing.

File: edu.cuny.hunter.hybridize.tests/resources/HybridizeFunction/testTargetedCfaDepthSoundness/in/A.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    # Set layers.
+    def __init__(self):
+        super(NeuralNet, self).__init__()
+        # First fully-connected hidden layer.
+        self.fc1 = layers.Dense(n_hidden_1, activation=tf.nn.relu)
+        # First fully-connected hidden layer.
+        self.fc2 = layers.Dense(n_hidden_2, activation=tf.nn.relu)
+        # Second fully-connecter hidden layer.
+        self.out = layers.Dense(num_classes)
```
[the remark this answers is not in the document]
Comment (khatchad) [reply edu.cuny.hunter.hybridize.tests/resources/HybridizeFunction/testTargetedCfaDepthSoundness/in/A.py:90]: Leaving these as-is. This fixture is a verbatim copy of the existing `testNeuralNetwork` fixture (itself vendored from aymericdamien/TensorFlow-Examples); the two `A.py` files are byte-identical. The `fully-connecter` typo is upstream's, and the labels are actually correct (the comments trail their statements: `fc1` is the first layer, `fc2` the second). Editing them would diverge this copy from the sibling fixture and the upstream source for no behavioral effect.
